### ISSUE TITLE
Hack NIght Discussion

### ISSUE THREAD

**Issue by AEgan** [rubyforgood/share_christmas#119]
Tonight at the Arlington Ruby meetup we discussed where the project currently stands and a way to narrow down scope to move the project forward. One of the problems we kept getting back to is that we started developing features backwards. The main feature we worked out outside of active admin was the point at which a member matches with a recipient. Here is a loose overview of everything that needs to happen before that point:
- a volunteer center starts a campaign
- a volunteer center creates an organization
- a volunteer center gets a list of recipient families, with recipients
- a volunteer center assigns a list of recipient families to an organization
- an organization adds members
- a member commits to a campaign

After all of that, a member can match with a recipient. The main problem was we never built out, either in code or in user stories, how anything before the match worked.

Here is the solution we decided on
1. Deprecate ActiveAdmin
   - ActiveAdmin allowed us to create a way for someone to do everything, but as a result, we didn't think about how anyone did anything at this foundation level.
   - Removing it will allow us to focus on a solid MVP by making us think about how each step should work. This should result in a more solid, scaleable, and maintainable process
2. Get rid of Volunteer Center management
   - Having multiple volunteer centers use this app would be a "nice to have", and therefore should be eliminated for an MVP.
   - This will also eliminate a user profile. We don't need to worry about what a volunteer center admin can or can't do. They can do everything, because we will start with 1 volunteer center. This is `admin@example.com`

This leads us to the three areas we should focus on
- Organization management
  - As a Volunteer Center, I should be able to manage everything about an organization
- Campaign Management
  - As a Volunteer Center, I should be able to manage everything about campaigns
- Recipient Family management
  - As a Volunteer Center, I should be able to manage everything about a recipient family
  - As a Volunteer Center, I should be able to add recipients to a recipient family

Things to think about for a second step:
- What can an organization admin do?
  - Do organization admins add their organization to a campaign? Or can only volunteer centers do that
- How do volunteer centers assign recipient families to an organization?

I will create issues for each of the user stories defined above, and would love to have a discussion about what an MVP should look like. I'd prefer to have that here, in this issue, so that we have an easily accessable way to refer to this discussion.

**Comment by craig-riecke** [comment]
@AEgan: Thanks for getting this "unstuck".  The Volunteer Center management features sound right-on-target.  

For the second step, as far as I can tell from Durham Volunteer Center:
- The Volunteer Center will assign an organization to a campaign.  That acts as sort of an approval process.
- The Volunteer Center will assign recipient families to an organization.   That makes it flexible for them to parcel out territory (if they know an org can handle that, they might give them more, and a smaller org less.)   At this point, it can be as easy as selecting recipient families by  checkbox, selecting an org from a drop down, and clicking "Assign".  We're not talking thousands of families here, so no fancy selection criteria stuff is needed for an MVP.

**Comment by atzorvas** [comment]
@AEgan 
Removing ActiveAdmin and Volunteer Center management shouldn't be issues on their own for the MVP and even better as a clean up before delivering the MPV part 1?

**Comment by craig-riecke** [comment]
All righty, it looks like we're in agreement that these features are good, so I'm gonna create individual issues for them.  I think @atzorvas had the idea of putting the Volunteer Center admin under a controller namespace, and that sounds good to me (we might follow suit with Org Center admin's too -- not sure yet).

**Comment by craig-riecke** [comment]
Oh two small things, which will get fleshed out in the individual tasks as well:
- Currently, organizations can ask to join a campaign as well - that's already on the Org Admin screen.  We just need to add an email to the VC to kick off the "assign recipient families to org" part.  (The VC can say "we've got too many orgs already" or "it's too late to join" or whatever in an out-of-band conversation.)
- Members do not really commit to a campaign.  They just get matched, and that's their commitment.  That makes the app slightly easier.
